File: fixtures/scale/java/large/src/cryptobench/ciphers/twosquare/TwoSquareSuite.java
package cryptobench.ciphers.twosquare;

import cryptobench.core.Cipher;
import cryptobench.core.CipherSuite;
import cryptobench.core.SuiteResult;
import cryptobench.verify.RoundTrip;

public class TwoSquareSuite implements CipherSuite {
    @Override
    public String name() {
        return "twosquare";
    }

    @Override
    public SuiteResult run() {
        Cipher cipher = new TwoSquareCipher(TwoSquareKey.defaultKey());
        int passed = 0;
        int failed = 0;
        long start = System.nanoTime();
        for (String sample : TwoSquareVectors.samples()) {
            if (RoundTrip.check(cipher, sample)) passed++; else failed++;
        }
        return new SuiteResult(name(), passed, failed, System.nanoTime() - start);
    }
}

File: fixtures/scale/kotlin/large/src/cryptobench/ciphers/twosquare/TwoSquareSuite.kt
package cryptobench.ciphers.twosquare

import cryptobench.core.Cipher
import cryptobench.core.CipherSuite
import cryptobench.core.SuiteResult
import cryptobench.verify.roundTrip

class TwoSquareSuite : CipherSuite {
    override fun name(): String = "twosquare"

    override fun run(): SuiteResult {
        val cipher: Cipher = TwoSquareCipher(TwoSquareKey.default())
        var passed = 0
        var failed = 0
        val start = System.nanoTime()
        for (sample in TwoSquareVectors.samples()) {
            if (roundTrip(cipher, sample)) passed++ else failed++
        }
        return SuiteResult(name(), passed, failed, System.nanoTime() - start)
    }
}
